src/components/menu/menu-profile.js

```diff
@@ -0,0 +1,77 @@
+import {
+  ListProfile as StyledListProfile,
+  ListItemProfile as StyledListItemProfile,
+  ProfileImgContainer as StyledProfileImgContainer,
+  ProfileImg as StyledProfileImg,
+  ProfileLabelsContainer as StyledProfileLabelsContainer,
+  ProfileTitle as StyledProfileTitle,
+  ProfileSubtitle as StyledProfileSubtitle,
+  ProfileBody as StyledProfileBody,
+} from './styled-components';
+import {mapStyletronProps} from './utils';
+import {getOverride} from '../../helpers/overrides';
+// Types
+import type {StatelessMenuProfilePropsT} from './types';
+
+export default function MenuProfile({
+  items,
+  getProfileItemLabels,
+  getProfileItemImg,
+  getProfileItemImgText,
+  getRequiredItemProps = (item, index) => ({key: String(index)}),
+  rootRef = React.createRef(),
+  overrides = {},
+}: StatelessMenuProfilePropsT) {
+  const ListProfile = getOverride(overrides.ListProfile) || StyledListProfile;
+  const ListItemProfile =
+    getOverride(overrides.ListItemProfile) || StyledListItemProfile;
+
+  const ProfileImgContainer =
+    getOverride(overrides.ProfileImgContainer) || StyledProfileImgContainer;
+  const ProfileImg = getOverride(overrides.ProfileImg) || StyledProfileImg;
+
+  const ProfileLabelsContainer =
+    getOverride(overrides.ProfileLabelsContainer) ||
+    StyledProfileLabelsContainer;
+  const ProfileTitle =
+    getOverride(overrides.ProfileTitle) || StyledProfileTitle;
+  const ProfileSubtitle =
+    getOverride(overrides.ProfileSubtitle) || StyledProfileSubtitle;
+  const ProfileBody = getOverride(overrides.ProfileBody) || StyledProfileBody;
+
+  return (
+    <ListProfile $ref={rootRef}>
+      {items.map((item, index) => {
+        const requiredProps = getRequiredItemProps(item, index);
+        // $FlowFixMe
+        const {key, ...itemProps} = mapStyletronProps(requiredProps);
+        const ItemImg = getProfileItemImg(item);
+        const {title, subtitle, body} = getProfileItemLabels(item);
+        return (
+          // Need to be explicit with `key` otherwise eslint throws error?
+          <ListItemProfile key={key} {...itemProps}>
+            <ProfileImgContainer>
+              {ItemImg &&
+                (typeof ItemImg === 'string' ? (
+                  // Render img src string wrapped with image component
+                  <ProfileImg src={ItemImg} alt={getProfileItemImgText(item)} />
+                ) : (
+                  // Or just render the entire component user specified
+                  // TODO: need better way of checking if valid React component
+                  <ItemImg />
+                ))}
+            </ProfileImgContainer>
+            <ProfileLabelsContainer>
+              {title && <ProfileTitle>{title}</ProfileTitle>}
+              {subtitle && <ProfileSubtitle>{subtitle}</ProfileSubtitle>}
+              {body && <ProfileBody>{body}</ProfileBody>}
+            </ProfileLabelsContainer>
+          </ListItemProfile>
+        );
+      })}
+    </ListProfile>
+  );
+}
```

Comment:
I could be missing something, but it seems like instead of exporting a `MenuProfile` we should just be exporting a `MenuProfileOption` that renders each row, and people pass that class to Menu/Select via overrides or a render prop.

```js
// Most people will be using Menu via Select (I think?)
import {Select, SelectOptionProfile} from 'baseui/select';

function ProfileOption(props) {
  return (
    <SelectOptionProfile
      {...props} 
      imageSrc={() => props.item.imageSrc}
      label={() => props.item.label}
      sublabel={() => props.item.subtitle}
    />
  );
}

export () => (
<Menu items={users} overrides={{Option: ProfileOption}} />
);
```

This might be more in line with how an end user would think about customizing the layout of a select/menu option.

But again, I might be missing something. How were you thinking the code would look for end users to utilize MenuProfile when using Select?

Reply:
They would import `MenuProfile` instead of `Menu`.

I had thought about the approach you mentioned, but I couldn't work out exactly how people would in-turn override the internals of `MenuProfile` without chaining multiple `overrides` props.

Reply:
>They would import MenuProfile instead of Menu.

Gotcha. If I'm using Select, do I have to import MenuProfile and then pass MenuProfile in as an override?

>I had thought about the approach you mentioned, but I couldn't work out exactly how people would in-turn override the internals of MenuProfile without chaining multiple overrides props.

Not sure if this helps, but tooltip has an example of passing along overrides to another nested component (it uses Popover internally) using the mergeOverrides helper:
https://github.com/uber-web/baseui/blob/master/src/components/tooltip/tooltip.js

Happy to jam on this together if it'd be helpful (i'm remote much of this week but could zoom).

Reply:
My hunch was that `MenuProfile` wouldn't exist, and we'd only export a `MenuProfileOption` that can be easily passed as an override to `Menu` or `Select`.

```js
<Menu overrides={{Option: MenuProfileOption}}>
```

If for some reason you need more control over mapping props or overrides, you could wrap MenuProfileOption:

```js
const CustomOptionRenderer = props => (
  <MenuProfileOption 
    {...props}
    label={props.option.user.name}
    imgSrc={props.option.user.avatar}
    overrides={{Label: {style: {color: 'red'}}}}
  />
);

<Menu overrides={{Option: CustomOptionRenderer}}>

// or with Select, which would just pass override to Menu:
<Select overrides={{Option: CustomOptionRenderer}}>
```

That's sort of the approach I was imagining, but there could definitely be some things I'm missing. Is there behavior that `MenuProfile` offers on top of this?

Reply:
Yep, I was concerned about exactly whether that would be a cleaner approach vs. just exporting a different top-level component. It seemed odd to be defining a new component just to override the internals, but maybe I'm over thinking it.

I can actually code it out and see how it looks though because I agree that it would make more sense contextually.

Reply:
Gotcha, yeah I see your thinking – definitely feels a bit strange at first exporting something just for use inside of overrides. Might feel a bit more natural as a render prop like `renderOption={profileOptionRenderer}`, but alas, we've sort of gravitated towards overrides for doing these kinds of things for better or worse.

One other thought why this strikes me as the right approach – if we come up with other standard option layouts in the future like `MenuFooOption`, `MenuBarOption`, etc, it might be a little bit redundant to have to also develop and export `MenuFoo`, `StatefulMenuFoo`, `MenuBar`, `StatefulMenuBar`.

But anyways, definitely try it out – there may be things I haven't considered.